src/app/counters/counters.page.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+
+import { Counter } from '../model/counter';
+import { CounterService } from '../model/counter.service';
+
+import { Instance } from '../model/instance';
+import { InstanceService } from '../model/instance.service';
+
+@Component({
+  selector: 'app-counters',
+  templateUrl: 'counters.page.html',
+  styleUrls: ['counters.page.scss']
+})
+export class CountersPage implements OnInit {
+  private selectedItem: any;
+
+  constructor(
+    private counterService: CounterService,
+    private instanceService: InstanceService
+  ) {
+  }
+
+  get counters() {
+    return this.counterService.all;
+  }
+
+  public addCounter() {
+    this.counterService.save(
+      new Counter()
+    );
+  }
+
+  public deleteCounter(counter: Counter) {
+    this.counterService.delete(counter);
+  }
+
+  public addInstance(counter: Counter, count: number) {
+    counter.count += count;
+    this.counterService.save(counter);
+    this.instanceService.save(
```

Comment:
dr zählerstand wird so dür `counter.count`, aber ou dür aui `Instance` iträg für dä counter abbiudet. dasme z total no separat füert fingi ok, aber i denke de sötte nid d GUI komponente verantwortlech si die beidi nachezfüehre. villecht chönntme das mit eim service statt zwöi mache?

Reply:
Agree, d gui soet nid d logik vom zeue implementiere. Aber I ha die beide services bewusst trennt, mit hinsicht das d visualisierige auae nume no dr instance service bruche. Witer moechti komplexitaet chli bhaute u di beide services nid vone nang abhangig mache. Villech e dritte service wo die beide angere services beinhautet?

Reply:
extracted this logic and moved it to the `CounterService.count()` method.